haystack/document_store/weaviate.py

```diff
@@ -0,0 +1,592 @@
+                                          added already exists.
+        :param return_embedding: To return document embedding.
+        :param embedding_field: Name of field containing an embedding vector.
+        :param progress_bar: Whether to show a tqdm progress bar or not.
+                             Can be helpful to disable in production deployments to keep the logs clean.
+        """
+
+        # save init parameters to enable export of component config as YAML
+        self.set_config(
+            weaviate_url=weaviate_url, timeout_config=timeout_config, username=username, password=password,
+            index=index, vector_dim=vector_dim, text_field=text_field, name_field=name_field,
+            faq_question_field=faq_question_field, meta_field=meta_field, similarity=similarity, index_type=index_type,
+            custom_schema=custom_schema, module_name=module_name, update_existing_documents=update_existing_documents,
+            return_embedding=return_embedding, embedding_field=embedding_field, progress_bar=progress_bar,
+        )
+
+        # Connect to Weaviate server using python binding
+        if username and password:
+            secret = AuthClientPassword(username, password)
+            self.weaviate_client = client.Client(url=weaviate_url,
+                                             auth_client_secret=secret,
+                                             timeout_config=timeout_config)
+        else:
+            self.weaviate_client = client.Client(url=weaviate_url,
+                                                 timeout_config=timeout_config)
+
+        # Test Weaviate connection
+        try:
+            status = self.weaviate_client.is_ready()
+            if not status:
+                raise ConnectionError(
+                    f"Initial connection to Weaviate failed. Make sure you run Weaviate instance "
+                    f"at `{weaviate_url}` and that it has finished the initial ramp up (can take > 30s)."
+                )
+        except Exception:
+            raise ConnectionError(
+                f"Initial connection to Weaviate failed. Make sure you run Weaviate instance "
+                f"at `{weaviate_url}` and that it has finished the initial ramp up (can take > 30s)."
+            )
+        self.index = index
+        self.vector_dim = vector_dim
+        self.text_field = text_field
+        self.name_field = name_field
+        self.faq_question_field = faq_question_field
+        self.meta_field = meta_field
+        self.similarity = similarity
+        self.index_type = index_type
+        self.custom_schema = custom_schema
+        self.module_name = module_name
+        self.update_existing_documents = update_existing_documents
+        self.return_embedding = return_embedding
+        self.embedding_field = embedding_field
+        self.progress_bar = progress_bar
+
+        self._create_schema_and_index_if_not_exist(self.index)
+
+    #def __del__(self):
```

Comment:
let's remove this leftover before we merge. or is there an equivalent connection that we need to close for weaviate?

Reply:
sure, removed this code.